src/api_puller.R

```diff
@@ -1,70 +1,113 @@
-# to do (j): document this function
-api_puller <- function(site, start_dt, end_dt = Sys.time(), api_token, dump_dir, require = NULL) {
+#' @title Download API data from HydroVu
+#'
+#' @description A function designed to download raw API data from HydroVu and store in a user-selected file location.
+#'
+#' @param site List of sites to download data for.
+#' @param start_dt Start DT for downloading data.
+#' @param end_dt End DT for downloading data. Default is the current time.
+#' @param api_token API token for accessing HydroVu. Pulled in via `hv_auth()`.
+#' @param dump_dir Directory where raw API data will be saved to.
 
-  locs <- hv_locations_all(api_token)
+api_puller <- function(site, network, start_dt, end_dt = Sys.time(), api_token, dump_dir) {
+
+  if(network %in% c("all", "All", "virridy", "Virridy")){
+
+    locs <- hv_locations_all(hv_token)
+
+  } else if(network %in% c("csu", "fcw", "CSU", "FCW")){
+
+    locs <- hv_locations_all(hv_token) %>%
+      dplyr::filter(!grepl("virridy", name, ignore.case = TRUE))
+
+  }
```

Comment:
Question: FC system will only have CSU/csu/FCW/fcw networks?

Reply:
Yep! All the Virridy stuff can and should get removed from this.